src/Http/Http.Abstractions/src/Extensions/UsePathBaseExtensions.cs

```diff
@@ -31,6 +32,17 @@ public static IApplicationBuilder UsePathBase(this IApplicationBuilder app, Path
             return app;
         }
 
+        const string globalRouteBuilderKey = "__GlobalEndpointRouteBuilder";
+        // Only use this path if there's a global router (in the 'WebApplication' case).
```

Comment:
Pretty sure we had the same question for some of the other middleware when we initially created this "global routing" concept. And the answer was, why not make it work if it's easy.

Reply:
> And the answer was, why not make it work if it's easy.

Then why didn't we make it work in the non-'WebApplication' case? This isn't feedback on the PR. I think we should leave it how you have it since it's a bit late to change that kind of behavior.

Reply:
non WAB doesn't have the global routing concept, and we don't have a way to effectively inject middleware before `UseRouting`.